grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcServiceBuilder.java

```diff
@@ -322,7 +322,112 @@ public GrpcServiceBuilder addService(String path, BindableService bindableServic
             return addService(path, interceptor, methodDescriptor);
         }
         registryBuilder.addService(path, bindableService.bindService(), methodDescriptor,
-                                   bindableService.getClass());
+                                   bindableService.getClass(), null);
+        return this;
+    }
+
+    /**
+     * Adds a gRPC {@link BindableService} with functions to this {@link GrpcServiceBuilder}.
+     * The functions will be applied in the specified order and before functions decorated with the service.
+     * In the following sample code, these functions will be applied
+     * in the order of FirstDecorator, SecondDecorator, ClassDecorator, and MethodDecorator.
+     * <pre>{@code
+     * @Decorator(ClassDecorator.class)
+     * class SimpleServiceImpl extends SimpleServiceImplBase {
+     *
+     *     @Decorator(MethodDecorator.class)
+     *     @Override
+     *     public void simpleCall(SimpleRequest request,
+     *                            StreamObserver<SimpleResponse> responseObserver) {
+     *         ...
+     *     }
+     * }
+     * ...
+     * GrpcService.builder()
+     *            .addService(new SimpleServiceImpl(),
+     *                        List.of(delegate -> delegate.decorate(new FirstDecorator()),
+     *                                delegate -> delegate.decorate(new SecondDecorator())))
+     *            .build()
+     * }</pre>
+     */
+    @UnstableApi
+    public GrpcServiceBuilder addService(
+            BindableService bindableService,
+            Iterable<? extends Function<? super HttpService, ? extends HttpService>> decorators) {
+        requireNonNull(bindableService, "bindableService");
+        if (bindableService instanceof ProtoReflectionService) {
+            final ServerServiceDefinition interceptor =
+                    ServerInterceptors.intercept(bindableService, newProtoReflectionServiceInterceptor());
+            return addService(interceptor);
+        }
+        requireNonNull(decorators, "decorators");
+        registryBuilder.addService(bindableService.bindService(), bindableService.getClass(), decorators);
+        return this;
+    }
+
+    /**
+     * Adds an implementation of gRPC service with functions to this {@link GrpcServiceBuilder}.
+     * @see #addService(Object, Function)
```

Comment:
Let's put an empty line before `@see`.
```suggestion
     * Adds an implementation of gRPC service with functions to this {@link GrpcServiceBuilder}.
     *
     * @see #addService(Object, Function)
```